WordPress/Classes/Models/Post+RefreshStatus.swift

```diff
@@ -6,27 +6,16 @@ extension Post {
     /// - Parameters:
     ///   - onCompletion: block to invoke when status update is finished.
     ///   - onError: block to invoke if any error occurs while the update is being made.
-    ///
-    static func refreshStatus(with coreDataStack: CoreDataStack, onCompletion: (() -> Void)? = nil, onError: ((Error) -> Void)? = nil) {
+    static func refreshStatus(with coreDataStack: CoreDataStack) {
         coreDataStack.performAndSave { context in
             let fetch = NSFetchRequest<Post>(entityName: Post.classNameWithoutNamespaces())
             let pushingPredicate = NSPredicate(format: "remoteStatusNumber = %@", NSNumber(value: AbstractPostRemoteStatus.pushing.rawValue))
             let processingPredicate = NSPredicate(format: "remoteStatusNumber = %@", NSNumber(value: AbstractPostRemoteStatus.pushingMedia.rawValue))
             fetch.predicate = NSCompoundPredicate(orPredicateWithSubpredicates: [pushingPredicate, processingPredicate])
-            let postsPushing = try context.fetch(fetch)
+            let postsPushing = (try? context.fetch(fetch)) ?? []
```

Comment:
Just and idea, have you considered returning early over coalescing to an empty array if the fetch fails?


```suggestion
            guard let postsPushing = try? context.fetch(fetch) else { return }
```

I don't think it makes a difference from an operation count point of view, besides llvm might be doing all sorts of optimizations here. So 🤷

Reply:
👍 thanks! I've taken your code suggestion.